dmapiclient/data.py

```diff
@@ -181,22 +181,48 @@ def set_framework_result(self, supplier_id, framework_slug, is_on_framework, use
             user=user,
         )
 
-    def register_framework_agreement_returned(self, supplier_id, framework_slug, user):
+    def register_framework_agreement_returned(self, supplier_id, framework_slug, user, signer_details=None):
+        framework_interest_dict = {
+            "agreementReturned": True,
+        }
+        if signer_details is not None:
+            framework_interest_dict['signerDetails'] = signer_details
+
+        return self._post_with_updated_by(
+            "/suppliers/{}/frameworks/{}".format(
+                supplier_id, framework_slug),
+            data={"frameworkInterest": framework_interest_dict},
+            user=user,
+        )
+
+    def unset_framework_agreement_returned(self, supplier_id, framework_slug, user):
```

Comment:
We should lose the unset methods for both dates. This will be a breaking change, requiring a major version number bump and an entry in the CHANGELOG detailing the breaking change. (Although the unset agreement returned at is only used in one script, and should never need to be used again, so this won't have any knock-on effect for our apps.)